lib/puppet/defaults.rb

```diff
@@ -1395,7 +1395,24 @@ def self.default_diffargs
     },
     :server => {
       :default => "puppet",
-      :desc => "The puppet master server to which the puppet agent should connect."
+      :desc => "The puppet master server to which the puppet agent should connect.",
+      :hook => proc { |value|
+          if Puppet.settings.set_by_config?(:server_list)
```

Comment:
What was the outcome of the deprecation discussion? Fwiw, my least favorite option is that we live with two different settings forever and ever. Allowing `server` to be multivalued would be one approach, but failing that, I'd favor deprecating `server` over having both settings forever.  If concern over deprecation is high, we could hold off on deprecating until we know we're on the last 4.y (i.e. once we decide when to do a puppet 5).

Thoughts?